KwinEsther/Journa
# Journa

Overview
This pull request introduces a Clarity-based blockchain voting contract named Journa. The contract allows users to vote for their favorite fruits from a predefined list. Each user is limited to a maximum of 5 votes, and votes are tracked both at the user level and the fruit level. Additionally, the contract provides several read-only functions to query the current votes for a specific fruit, check the number of votes cast by a user, and identify the leading fruit in real time.

Features
Vote for Fruits: Users can vote for fruits from a predefined list (apple, banana, cherry, etc.). A user can cast up to 5 votes.
Vote Tracking: Tracks the total number of votes for each fruit and how many votes each user has cast.
Leading Fruit: Provides a way to query the fruit with the most votes.
Read-Only Functions:
get-fruit-votes: Retrieve the current vote count for a given fruit.
get-fruits: Retrieve the list of available fruits.
get-user-vote-count: Retrieve the number of votes cast by a specific user.
get-leading-fruit: Retrieve the fruit with the highest number of votes.
Contract Details
Data Variables:
fruits: List of fruits available for voting.
fruit-votes: Map to store the number of votes for each fruit.
user-votes: Map to track the number of votes each user has cast.
MAX_VOTES_PER_USER: Constant that limits users to a maximum of 5 votes.
Main Functions:
vote-for-fruit: Allows users to vote for a fruit, checks if the user has reached their vote limit, and updates the fruit’s vote count.
Helper functions to track votes and ensure users do not exceed the vote limit.
Changes in This PR
Added data variables for fruits, votes, and user vote tracking.
Implemented a voting mechanism where each user can cast votes on fruits.
Created read-only functions to fetch votes, fruits, user vote counts, and the leading fruit.
Defined the constant MAX_VOTES_PER_USER to limit voting behavior.
How to Test
Deploy the contract to the Stacks blockchain.
Interact with the contract by calling vote-for-fruit to cast votes on fruits.
Use get-fruit-votes to verify vote counts for each fruit.
Call get-user-vote-count to verify the number of votes cast by a specific user.
Use get-leading-fruit to get the current leading fruit based on votes.
Related Issues
N/A
Notes
The contract ensures that users cannot vote more than the maximum allowed votes (MAX_VOTES_PER_USER).
The contract is designed for simplicity with basic features focused on voting and vote tracking.
